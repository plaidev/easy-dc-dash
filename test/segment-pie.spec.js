import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import SegmentPie from '@/segment-pie.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('segment-pie', () => {
  const segments = {
    all: 'all'
  };
  Store.registerData([
    { segments: 'all', d1: 'a', d2: true, v: 1 },
    { segments: 'all', d1: 'a', d2: false, v: 2 },
    { segments: 'all', d1: 'b', d2: false, v: 3 },
    { segments: 'all', d1: 'b', d2: true, v: 5 }
  ]);

  const wrapper = mount(localVue.extend(SegmentPie), {
    localVue,
    propsData: {
      id: 'test-segment-pie',
      dimension: 'd1',
      reduce: 'v',
      segments
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
