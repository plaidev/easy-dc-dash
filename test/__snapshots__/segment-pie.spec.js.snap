// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`segment-pie to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-segment-pie" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g transform="translate(100,100)" class="">
                            <g class="pie-slice-group">
                                <g class="pie-slice _0">
                                    <path fill="#3182bd" d="M0,100A100,100 0 1,1 0,-100A100,100 0 1,1 0,100Z"></path>
                                </g>
                            </g>
                            <g class="pie-label-group">
                                <text class="pie-slice pie-label _0">none</text>
                            </g>
                        </g>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
